### ISSUE TITLE
"Unable to load the page" ☂  bug

### ISSUE THREAD

**Issue by paulirish** [GoogleChrome/lighthouse#2784]
Within Lighthouse, this error comes from `gatherRunner.assertPageLoaded`. This is called right after we finish loading the page for a pass (there are 3 passes by default). It looks for the HTML resources's network record (via some URL comparison). 

If it fails to find a match, we error. Or if we find a match, but the netword record is marked as `failed`, then we error.

There are a few cases in which this error shows up.

1. **Server failure with HTTP URL:** The HTML request was matched but determined to be failed. This happens with incorrect server configuration. Example: #2459 (redirect configuration was botched, resulting in a error page if you navigate to the HTTP regularly in chrome)
1. **Server doesn't offer HTTP:** The host doesn't offer a site at the HTTP URL at all. Example: #1978
1. **HSTS:** The host is using HSTS, and Chrome can't allow the browser to attempt to navigate to HTTP. Example: #2465
1. **about:blank:** Audits was started on about:blank somehow. Canonical issue: #2362. This was [fixed](https://github.com/ChromeDevTools/devtools-frontend/commit/a3b3d85785aee95f86b489c81bc8a63c1f011dcb) on the DevTools side, and will ship in m61.

**Comment by zhanjh** [comment]
I got this bug "#3199" because my server doesn't offer the right HTTP URL. This is my old nginx config:
```
server {
    listen    80;
    server_name    *.tecposter.cn;
    return 301 https://$server_name$request_uri;
}
```
The value of $server_name is "*.tecposter.cn" and it need to be changed to $http_host:

```
server {
    listen    80;
    server_name    *.tecposter.cn;
    return 301 https://$server_name$http_host;
}
```

**Comment by patrickhulce** [comment]
Assigning @brendankenny for fixing the #2459 related instances

**Comment by ockxdd** [comment]
https://guides.github.com/features/mastering-markdown/

**Comment by clubclick** [comment]
the way i got around this issue was to run each audit option at a time instead of all at once

**Comment by nhoizey** [comment]
@clubclick even if I run only the PWA audit, it doesn't work on [my website](https://nicolas-hoizey.com/)

**Comment by techy16** [comment]
I have opened the issue #4884 which is showing closed now as it is showing duplicate. Do you have any resolutions to this issue?

LHError: FAILED_DOCUMENT_REQUEST
at Function.getPageLoadError (/usr/local/lib/node_modules/lighthouse/lighthouse-core/gather/gather-runner.js:163:21)
at pass.then._ (/usr/local/lib/node_modules/lighthouse/lighthouse-core/gather/gather-runner.js:298:36)
at
at process._tickCallback (internal/process/next_tick.js:188:7)

**Comment by KolyaSlisarenko** [comment]
Hello, there is a lot of link in google that follows to this page, that shows a lot of cases of the error but without examples of solution of the problem. Could you provide some examples to fix these issues quickly?

**Comment by patrickhulce** [comment]
@KolyaSlisarenko the examples are outlined in the issue description as 1-6. Do none of those cases apply to you?

**Comment by KolyaSlisarenko** [comment]
@patrickhulce I ran lighthouse using sudo command and it works) It will be interesting how long it will work

**Comment by nitinmobikasa1234** [comment]
Hi,
Can anyone tell me how I can resolve this "Lighthouse" error.
URL: https://www.izenda.com/
And how I can analyze the cause of error in the website.

**Comment by paulirish** [comment]
Let's turn the top of this issue into an FAQ/help page.

**Comment by SamB** [comment]
I assume that, once the page is ready, the issue description will get a nice big link to the new page at the top and/or bottom?  Because there are a LOT of things that link here.

**Comment by aaaroaaa** [comment]
Hi. I still have this error (as described here - https://github.com/GoogleChrome/lighthouse/issues/9228). But it seems to me that none of the described items is a root cause of the error.

**Comment by thomwoz** [comment]
Hello,
my issue is very similar. I can successfully perform test using Pagespeed UI or when I request data from my application run locally, but when try to perform test from my AWS server, pagespeed responds with ERRORED_DOCUMENT_REQUEST. 
So I guess there is some issue on my AWS server, since I am able to get the results requesting from other places. 
Do you have any idea what might be wrong?

**Comment by vittoriograssi** [comment]
Hello everybody,
I have the same problem with PageSpeed on the domain:
www.smarshop.it

Can you help me understand what the problem may be?
I also noticed that Google Analytics also fails to intercept the traffic on this website.

**Comment by naveenrajbu** [comment]
Hi guys, my site http://ezyleads.in continues to show status 403 Errored_document_request but it works for http://ezyleads.in/index.php
Help me to solve this problem.

**Comment by Stefadesign** [comment]
So is there any solutions for issue. I am getting the same error ?

**Comment by rahultripathi15** [comment]
I have a website where I have places Google Analytics code in Header. Initially, this property got verified and working fine.

After about a month this property itself got not verified. I have tried all possible ways including,

Verification by Analytics code
Verification by Google Tag
Verification by HTML Tag
and none of these methods worked for me.

When I tried testing my URL on Google Page Speed Test - It shows the following error:

Lighthouse returned an error: FAILED_DOCUMENT_REQUEST. Lighthouse was unable to reliably load the page you requested. Make sure you are testing the correct URL and that the server is properly responding to all requests. (Details: net::ERR_CONNECTION_FAILED)

Others are saying my website got penalized and it is blocked by Google. This issue will not fix.

Please help; what is the issue? How I can fix this?

My website URL is: https://bestringtones-songlyrics.com

and Speed text URL is:
https://developers.google.com/speed/pagespeed/insights/?url=http%3A%2F%2Fbestringtones-songlyrics.com%2F

Please guide.

![isssue 1](https://user-images.githubusercontent.com/61452542/75231525-98b9f700-57db-11ea-8df0-b0696c98fba0.JPG)

**Comment by rahultripathi15** [comment]
Why this error.

![issue 2](https://user-images.githubusercontent.com/61452542/75231907-4d541880-57dc-11ea-9720-e61c67de6329.JPG)

**Comment by gamooga-echoabhishek** [comment]
Same here few days back lighthouse audits were working fine.
Today I checked it was not working. Getting the same error as rahul's.

Website link: https://as2dc10.gamooga.com:456/
Error Mesage: 
Error: Error: Lighthouse returned error: FAILED_DOCUMENT_REQUEST. Lighthouse was unable to reliably load the page you requested. Make sure you are testing the correct URL and that the server is properly responding to all requests. (Details: net::ERR_FAILED)

**Comment by j8n** [comment]
I had the same issue 'ERRORED_DOCUMENT_REQUEST' with the 503 error code on a wordpress site! The problem was the wordfence security plugin for me. When I deactivated the plugin the problem fixed.

**Comment by ferrerluis** [comment]
I wanted to drop a comment and mention this `ERRORED_DOCUMENT_REQUEST` error happens when scanning Google, too, and even for different countries

[Google US (i.e. `.com`)](https://www.googleapis.com/pagespeedonline/v5/runPagespeed?url=https://www.google.com/)
[Google Netherlands (i.e. `.de`)](https://www.googleapis.com/pagespeedonline/v5/runPagespeed?url=https://www.google.de/)

I don't believe Google matches the criteria above, so I am unsure why it'd be doing that

**Comment by patrickhulce** [comment]
@ferrerluis Google case usually falls under "Server failure with initial request URL". Google servers typically reject a portion of bot traffic and have a fairly low rate limit, so if depending on how you're requesting or if you're using a shared service you can hit this fairly frequently.

**Comment by jamesarosen** [comment]
Our Lighthouse logs end with

> Runtime error encountered: Lighthouse was unable to reliably load the page you requested. Make sure you are testing the correct URL and that the server is properly responding to all requests. (Status code: 503)

but I've checked our CDN logs and we haven't served a 503 to Lighthouse in the relevant time-frame. It's certainly possible that a sub-resource returns a 503, but I can't tell which one because the logs don't specify which URL is the problem.

**Comment by patrickhulce** [comment]
503 means that the service is unavailable. Is it possible that whatever is handling the request could go down in a way that wouldn't appear in the logs you're looking at?

This error is only given when Lighthouse receives an error status code for the root document. A 503 for a subresource would not trigger this error.

**Comment by jamesarosen** [comment]
> Is it possible that whatever is handling the request could go down in a way that wouldn't appear in the logs you're looking at?

That's a good question. [SpeedCurve](https://speedcurve.com) is running Lighthouse for us in this case. I'll check with them.

> This error is only given when Lighthouse receives an error status code for the root document.

That's very helpful! Thank you.

**Comment by Ninetheme** [comment]
For WordPress users, check svg http:// urls in your pages, safe svg plugin insert http:// and it will slow down your pages. reported

**Comment by charlesmorgan40** [comment]
> Within Lighthouse, this error comes from `gatherRunner.assertPageLoaded`. This is called right after we finish loading the page for a pass (there are 3 passes by default). It looks for the HTML resources's network record (via some URL comparison).
> 
> If it fails to find a match, we error. Or if we find a match, but the network record is marked as `failed`, then we error.
> 
> There are a few cases in which this error shows up.
> 
> 1. **Server failure with initial request URL:** The HTML request was matched but determined to be failed. This happens with incorrect server configurations, when the server does not reply at all, or replies with an empty/otherwise invalid response. Examples: #2459 (redirect configuration was botched, resulting in a error page if you navigate to the HTTP regularly in chrome) #2500 (TLS handshake error) #2646 (a local server replies with an empty response)
>    **Solution:** Fix your server to return valid HTML documents with successful status codes.
> 2. **Server doesn't offer HTTP:** The host doesn't offer a site at the HTTP URL at all. Example: #1978
>    **Solution:** Change your server to redirect HTTP traffic to HTTPS
> 3. ~**HSTS:** The host is using HSTS, and Chrome can't allow the browser to attempt to navigate to HTTP. Example: #2465~
> 4. **about:blank:** Audits was started on about:blank somehow. Canonical issue: #2362. This _should_ be [fixed](https://github.com/ChromeDevTools/devtools-frontend/commit/a3b3d85785aee95f86b489c81bc8a63c1f011dcb) on the DevTools side, and will ship in m61, but can still happen occasionally for unknown reasons. If this happens _consistently_ on a particular URL for you, please file a separate issue specifying the URL.
>    **Solution:** Restart Chrome and run Lighthouse again. If problem persists with a particular URL, file a new issue here with the title "Reproducible about:blank errors"
> 5. **No document request found:** Lighthouse was unable to identify any network request as the main HTML resource. Example: #3496
>    **Solution:** Restart Chrome and run Lighthouse again. If problem persists with a particular URL, file a separate issue with the title "Cannot identify main document"
> 6. **Bad TLS/certificate interstitial.** Lighthouse couldn't load the page because Chrome couldn't handle the security configuration of the URL. Example #1123
>    **Solution:** Fix your server to return a valid, secure TLS certificate.

**Comment by rahultripathi15** [comment]
Hi,
Thank you so much for the continuous update.

Thanks

**Comment by patrickhulce** [comment]
The outstanding work here is to take the body of the issue and add it to an FAQ in our readme.

**Comment by padam2403** [comment]
Hi,

While using the Google Pagespeed v5 Lighthouse API, I generally face the HTTP 500 Error.

Can anyone help how can I overcome this?

Thanks,
Padam

**Comment by aseem-upadhyay** [comment]
In my case, was checking a few urls, which to my surprise works fine on web.dev / page speed insights. 

But, when I run the test via cli in a dockerised environment, lighthouse returns me the following errors (although works fine on my local docker instance)
```
 GatherRunner:error Lighthouse was unable to reliably load the page you requested. Make sure you are testing the correct URL and that the server is properly responding to all requests. (Status code: 403) https://www.magicbricks.com/prestige-lakeside-habitat-whitefield-bangalore-pdpid-4d4235303139343638
```
another one
```
GatherRunner:error Lighthouse was unable to reliably load the page you requested. Make sure you are testing the correct URL and that the server is properly responding to all requests. (Status code: 417) https://www.99acres.com/search/property/buy/residential-all/gurgaon?search_type=QS&refSection=GNB&search_location=CP1&lstAcn=CP_R&lstAcnId=1&src=CLUSTER&preference=S&selected_tab=1&city=8&res_com=R&property_type=R&isvoicesearch=N&keyword_suggest=gurgaon%3B&fullSelectedSuggestions=gurgaon&strEntityMap=W3sidHlwZSI6ImNpdHkifSx7IjEiOlsiZ3VyZ2FvbiIsIkNJVFlfOCwgUFJFRkVSRU5DRV9TLCBSRVNDT01fUiJdfV0%3D&texttypedtillsuggestion=gurgaon&refine_results=Y&Refine_Localities=Refine%20Localities&action=%2Fdo%2Fquicksearch%2Fsearch&suggestion=CITY_8%2C%20PREFERENCE_S%2C%20RESCOM_R&searchform=1&price_min=null&price_max=null
```

Could someone help identify whether the problem lies in how it is being executed, or there is something wrong with the page itself.

**Comment by Bryze** [comment]
In my case, was checking a few urls, which to my surprise works fine on web.dev / page speed insights. 

But, when I run the test via cli in a dockerised environment, lighthouse returns me the following errors (although works fine on my local docker instance)
```
 GatherRunner:error Lighthouse was unable to reliably load the page you requested. Make sure you are testing the correct URL and that the server is properly responding to all requests. (Status code: 403) https://www.magicbricks.com/prestige-lakeside-habitat-whitefield-bangalore-pdpid-4d4235303139343638
```
another one
```
GatherRunner:error Lighthouse was unable to reliably load the page you requested. Make sure you are testing the correct URL and that the server is properly responding to all requests. (Status code: 417) https://www.99acres.com/search/property/buy/residential-all/gurgaon?search_type=QS&refSection=GNB&search_location=CP1&lstAcn=CP_R&lstAcnId=1&src=CLUSTER&preference=S&selected_tab=1&city=8&res_com=R&property_type=R&isvoicesearch=N&keyword_suggest=gurgaon%3B&fullSelectedSuggestions=gurgaon&strEntityMap=W3sidHlwZSI6ImNpdHkifSx7IjEiOlsiZ3VyZ2FvbiIsIkNJVFlfOCwgUFJFRkVSRU5DRV9TLCBSRVNDT01fUiJdfV0%3D&texttypedtillsuggestion=gurgaon&refine_results=Y&Refine_Localities=Refine%20Localities&action=%2Fdo%2Fquicksearch%2Fsearch&suggestion=CITY_8%2C%20PREFERENCE_S%2C%20RESCOM_R&searchform=1&price_min=null&price_max=null
```

Could someone help identify whether the problem lies in how it is being executed, or there is something wrong with the page itself.

**Comment by pattarika** [comment]
Same issue with Brzye.  Keeping getting 503 inside docker.
 "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/90.0.4414.0 Safari/537.36",
  "environment": {
    "networkUserAgent": "Mozilla/5.0 (Linux; Android 7.0; Moto G (4)) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4143.7 Mobile Safari/537.36 Chrome-Lighthouse",
    "hostUserAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/90.0.4414.0 Safari/537.36",
    }
  },
  "lighthouseVersion": "7.1.0",
  "runWarnings": [
    "Lighthouse was unable to reliably load the page you requested. Make sure you are testing the correct URL and that the server is properly responding to all requests. (Status code: 503)"
  ],

**Comment by SamB** [comment]
Hmm. it would be easier to figure out what's going on if lighthouse could show you the failing request (if any), rather than just the status code.

**Comment by clubclick** [comment]
The HyperText Transfer Protocol (HTTP) 500 Internal Server Error
reset service workers

**Comment by clubclick** [comment]
https://play.google.com/store/apps/details?id=com.app.buckinghampalce

**Comment by pattarika** [comment]
@SamB I call this command inside docker  
> #!/bin/bash
> 
> while IFS= read -r line
> do
>     echo $( date +%X ) 'Creating report for ' $line
>     lighthouse --quiet --no-enable-error-reporting --skip-audits redirects-http \
>     --chrome-flags="--headless --disable-gpu --no-sandbox --disable-dev-shm-usage --ignore-certificate-errors" \
>     $line --output html --output json
> done < 'urls.txt'

**Comment by zqbe** [comment]
My ticket has been closed automatically. but I'm still having the below issue.

When using PageSpeed for testing the site below, we got an error.

https://developers.google.com/speed/pagespeed/insights/?url=https%3A%2F%2Fwww.qbe.com%2Fau

Lighthouse returned error: FAILED_DOCUMENT_REQUEST. Lighthouse was unable to reliably load the page you requested. Make sure you are testing the correct URL and that the server is properly responding to all requests. (Details: net::ERR_CONNECTION_FAILED)

But the site is accessible. The wired thing is if we use PageSpeed from US region then it works fine. However, it's not working in Australia region.

**Comment by mreduar** [comment]
@zqbe how do you change the region?

**Comment by Mbellsudteen** [comment]
ในวันที่ ศ. 1 ต.ค. 2021 07:33 น. Eduar Bastidas ***@***.***>
เขียนว่า:

**Comment by PachVerb** [comment]
i just run 
```lighthouse --help```

tips Error:

```
ighthouse --help                                ✔  10030  22:17:37
internal/process/esm_loader.js:74
    internalBinding('errors').triggerUncaughtException(
                              ^

Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only file and data URLs are supported by the default ESM loader. Received protocol 'node:'
    at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:751:11)
    at Loader.resolve (internal/modules/esm/loader.js:85:40)
    at Loader.getModuleJob (internal/modules/esm/loader.js:229:28)
    at ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:51:40)
    at link (internal/modules/esm/module_job.js:50:36) {
  code: 'ERR_UNSUPPORTED_ESM_URL_SCHEME'
}
```
who can help me?

**Comment by PachVerb** [comment]
i want to know what happen?

**Comment by jigneshbhavani** [comment]
@PachVerb  Updating my node version to 16.0 resolved the same issue.

**Comment by PachVerb** [comment]
> > @PachVerb Updating my node version to 16.0 resolved the same issue.
> 
> > i just run `lighthouse --help`
> > tips Error:
> > ```
> > ighthouse --help                                ✔  10030  22:17:37
> > internal/process/esm_loader.js:74
> >     internalBinding('errors').triggerUncaughtException(
> >                               ^
> > 
> > Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only file and data URLs are supported by the default ESM loader. Received protocol 'node:'
> >     at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:751:11)
> >     at Loader.resolve (internal/modules/esm/loader.js:85:40)
> >     at Loader.getModuleJob (internal/modules/esm/loader.js:229:28)
> >     at ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:51:40)
> >     at link (internal/modules/esm/module_job.js:50:36) {
> >   code: 'ERR_UNSUPPORTED_ESM_URL_SCHEME'
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > who can help me?
> 
> @PachVerb Updating my node version to 16.0 resolved the same issue.

ok, work for me. thanks guys

**Comment by Mannyflow7** [comment]
> 
304 Not Modified
Access-Control-Allow-Origin: *
Age: 2318192
Cache-Control: public, max-age=315360000
Connection: keep-alive
Date: Mon, 18 Jul 2016 16:06:00 GMT
Server: Apache
Vary: Accept-Encoding
Via: 1.1 3dc30c7222755f86e824b93feb8b5b8c.cloudfront.net (CloudFront)
X-Amz-Cf-Id: TOl0FEm6uI4fgLdrKJx0Vao5hpkKGZULYN2TWD2gAWLtr7vlNjTvZw==
X-Backend-Server: developer6.webapp.scl3.mozilla.com
X-Cache: Hit from cloudfront
X-Cache-Info: cached

**Comment by adamraine** [comment]
I dug through the history of this issue and it does seem obsolete. However the last triage was this comment https://github.com/GoogleChrome/lighthouse/issues/2784#issuecomment-486856089 indicating we should create some permanent FAQ for this. So there is still work to do, but no bugs to fix and at this point there are few enough issues related to main document errors that we can handle them on a case-by-case basis.

**Comment by air-play-1** [comment]
كيف اقوم بتحسين الاداء [@air-play-1 ](https://air-play-1.com/)

**Comment by air-play-1** [comment]
https://air-play-1.com/
